service/storage_proxy.cc

```diff
@@ -3443,9 +3443,11 @@ class abstract_read_executor : public enable_shared_from_this<abstract_read_exec
             return _proxy->query_mutations_locally(_schema, cmd, _partition_range, timeout, _trace_state);
         } else {
             tracing::trace(_trace_state, "read_mutation_data: sending a message to /{}", ep);
+            _proxy->mark_sent(ep, clock_type::now());
```

Comment:
Hm, in case of detecting replica failures in order to potentially fail pending user requests it doesn't sound particularly bad to use the same resolution as for timeouts. After all, that's the resolution at which users already receive feedback if their queries timed out. I think that if the need for separate resolutions ever arises, it would be simple to just split then. After all, this typedef was named `clock_type`, which suggests a general-purpose clock for storage proxy if you're not interested in anything more specific. I think that if timeouts or any other modules want a de clock, they should define `timeout_clock_type` and use it.